docs/technical-design/feature-brief-session-expiration.md

```diff
@@ -0,0 +1,38 @@
+# Feature brief: Session Expiration
+
+## Introduction
+
+We use a third-party authentication provider ([IDM](https://confluenceent.cms.gov/display/IDM/IDM+Trainings+and+Guides)) which automatically logs out sessions due to inactivity after about 30 minutes. We don't have direct access to their timekeeping, but we know this is about how long they allow a session to be inactive. Thus, we manually track sessions internally for the same time period.
+
+## Expected behavior
+ Two minutes before the session will expire due to inactivity we show a warning modal. The modal displays a live countdown and has CTA buttons with 1. the ability to log out immediately 2. the ability to extend the session. This helps us fulfill accessibility requirements around [WCAG 2.2.1 Timing Adjustable](https://www.w3.org/WAI/WCAG21/Understanding/timing-adjustable.html).
+
+![session expiration modal](../../.images/session-expiring.png)
+
+### Possible outcomes after the session expiration modal is displayed:
+1. If the user chooses to logout, we redirect to landing page.
+2. If the user extends the session, we refresh their tokens and restart our counter for the session.
+3. If the user takes no action and the browser is still active, the 2 minute countdown will complete and the user will be automatically logged out and a error banner will appear on the landing page notifying them what happened.
+
+![session expired banner - relevant for outcome 3](../../.images/session-expired-banner.png)
+
+This auto logout behavior will happen even if the browser tab is in the background. Either way we are keep track of time and the session expiration flow will work as expected.
+
+## Known edge cases
```

Comment:
Do we know if or when we will tackle these edge cases? If there are tickets, then we should link them here.